cpp/csp/core/SRMWLockFreeQueue.h

```diff
@@ -14,7 +14,7 @@ namespace csp
   one of the simplest!  template type is required to have an intrinsic next pointer
 */
 template< typename T >
-class alignas(CACHELINE_SIZE) SRMWLockFreeQueue
+class CSP_PUBLIC alignas(CACHELINE_SIZE) SRMWLockFreeQueue
```

Comment:
Looks like some of these are missing the `Platform.h` include leading to the syntax errors in compilation

Reply:
No, the header propagates in from `csp/core/TIme.h` in this case. There's some weird behavior with the combination of `alignas` and `CSP_PUBLIC`, although locally it builds fine for me.

Reply:
ITs good practice to always include all of your deps directly, otherwise you can break if the header you rely on to transitively include stops including your dep.
As for the error, I havent seen which build failed but if its macos it could be the compiler

Reply:
Included all the headers btw